.travis.yml

```diff
@@ -21,55 +21,46 @@ env:
     - RUST_NIGHTLY_VERSION=nightly-2018-06-29
     - RUST_CLIPPY_VERSION=0.0.211
     - EJB_RUST_BUILD_DIR="$TRAVIS_BUILD_DIR/exonum-java-binding-core/rust/"
+    - SODIUM_BUILD=1
+    - ROCKSDB_BUILD=1
+    - SNAPPY_BUILD=1
+    - SKIP_COMPILE=true
+  matrix:
+    - CHECK_RUST=true
+    - CHECK_RUST=false
+# force building instead of using from apt.
+
+# run java tests for jdk8&jdk10 without rust checks, 
+# check rust code with jdk8 without running java tests
+matrix:
+  exclude:
+  - jdk: openjdk10
+    env: CHECK_RUST=true
 
 cache:
   directories:
     - "$HOME/.cargo"
     - "$HOME/.m2"
 
 before_install:
-    # Skip the build if only md files were updated.
+  # Skip the build if only md files were updated.
   - source skip-ci.sh
   - export PATH="$PATH":"$HOME/.cargo/bin"
-    # Install rustup if it's not already installed (i.e., not in CI cache).
+  # Install rustup if it's not already installed (i.e., not in CI cache).
   - which rustup > /dev/null || curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain "$RUST_COMPILER_VERSION"
```

Comment:
Is it a no-op for `CHECK_RUST` and populated cache?

Reply:
what do you mean?

Reply:
Will it install rust compiler every build or will it use cached one

Reply:
It will install rust compiler v.1.26.2 if there isn't any